packages/eslint-plugin/docs/rules/boolean-literal-compare.md

```diff
@@ -0,0 +1,34 @@
+# Flags unnecessary equality comparisons against booleans (boolean-literal-compare)
+
+Comparing boolean values to boolean literals is unnecessary, as those expressions will result in booleans too.
+Just use the boolean values directly or negate them.
+
+## Rule Details
+
+Examples of **incorrect** code for this rule
+
+```ts
+if (someCondition === true) {
+  // ...
+}
+
+if (checkCondition() == false) {
+  // ..
+}
+```
+
+Examples of **correct** code for this rule
+
+```ts
+if (someCondition) {
+  // ...
+}
+
+if (!checkCondition()) {
+  // ...
+}
+```
```

Comment:
This doesn't communicate what the rule does too well. I think it would be better with explicit types on the variables, so you can clearly show the valid and invalid cases

i.e. something like (feel free to flesh out these examples further and fix them up if they're wrong)

**incorrect**

```ts
declare const someCondition: boolean;
if (someCondition === true) {
}
```

**correct**

```ts
declare const someCondition: boolean;
if (someCondition) {
}

declare const someObjectBoolean: boolean | Record<string, unknown>;
if (someObjectBoolean === true) {
}

declare const someStringBoolean: boolean | string;
if (someStringBoolean === true) {
}

declare const someUndefinedCondition: boolean | undefined;
if (someUndefinedCondition === false) {
}
```